pkg/prometheus/promcfg.go

```diff
@@ -1020,6 +1035,12 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 		})
 	}
 
+	// Add user extra relabelings if specified.
+	extraRelabelings := cg.GetScrapeClassExtraRelabelings(scrapeClass)
+	if len(extraRelabelings) > 0 {
+		relabelings = append(relabelings, generateRelabelConfig(extraRelabelings)...)
+	}
+
```

Comment:
I wonder if we should log or warn through admission webhooks if we have clashing relabeling rules in the ScrapeClass and the underlying resource.

I guess we can start with a log and iterate with validating webhooks later?

Reply:
I think this might then become an issue for use case where it actually make sense to override some custom fields though (not that I have one in mind)

Reply:
I agree that a "merging" strategy would be hard to get right. My expectation would be that the scrape class relabelings apply first then the service/pod monitor's relabelings. The scrape class relabelings may delete a discovered label that the monitor relabelings rely on but I feel like it's a corner case and not something that the operator could/should prevent.